reactor-core/src/main/java/reactor/core/publisher/UnicastManySinkNoBackpressure.java

```diff
@@ -0,0 +1,199 @@
+ */
+package reactor.core.publisher;
+
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+
+import org.reactivestreams.Subscription;
+import reactor.core.CoreSubscriber;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Sinks.Emission;
+import reactor.util.context.Context;
+
+final class UnicastManySinkNoBackpressure<T> extends Flux<T> implements Sinks.Many<T>, Subscription, Contextable {
+
+	public static <E> UnicastManySinkNoBackpressure<E> create() {
+		return new UnicastManySinkNoBackpressure<>();
+	}
+
+	enum State {
+		INITIAL,
+		SUBSCRIBED,
+		TERMINATED,
+		CANCELLED,
+	}
+
+	volatile State state;
+
+	@SuppressWarnings("rawtypes")
+	private static final AtomicReferenceFieldUpdater<UnicastManySinkNoBackpressure, State> STATE = AtomicReferenceFieldUpdater.newUpdater(
+			UnicastManySinkNoBackpressure.class,
+			State.class,
+			"state"
+	);
+
+	private volatile CoreSubscriber<? super T> actual = null;
+
+	volatile long requested;
+	@SuppressWarnings("rawtypes")
+	static final AtomicLongFieldUpdater<UnicastManySinkNoBackpressure> REQUESTED =
+			AtomicLongFieldUpdater.newUpdater(UnicastManySinkNoBackpressure.class, "requested");
+
+	UnicastManySinkNoBackpressure() {
+		STATE.lazySet(this, State.INITIAL);
+	}
+
+	@Override
+	public Flux<T> asFlux() {
+		return this;
+	}
+
+	@Override
+	public void subscribe(CoreSubscriber<? super T> actual) {
+		Objects.requireNonNull(actual, "subscribe");
+
+		if (!STATE.compareAndSet(this, State.INITIAL, State.SUBSCRIBED)) {
+			Operators.reportThrowInSubscribe(actual, new IllegalStateException("Unicast Sinks.Many allows only a single Subscriber"));
+			return;
+		}
+
+		this.actual = actual;
+		actual.onSubscribe(this);
+	}
+
+	@Override
+	public void request(long n) {
+		if (Operators.validate(n)) {
+			Operators.addCap(REQUESTED, this, n);
+		}
+	}
+
+	@Override
+	public void cancel() {
+		switch (STATE.getAndSet(this, State.CANCELLED)) {
+			case SUBSCRIBED:
+				actual = null;
+				break;
+		}
+	}
+
+	@Override
+	public Context currentContext() {
+		CoreSubscriber<? super T> actual = this.actual;
+		return actual != null ? actual.currentContext() : Context.empty();
+	}
+
+	@Override
+	public void emitNext(T value) {
+		switch (tryEmitNext(value)) {
+			case FAIL_OVERFLOW:
+				Operators.onDiscard(value, currentContext());
+				//the emitError will onErrorDropped if already terminated
+				emitError(Exceptions.failWithOverflow("Backpressure overflow during Sinks.Many#emitNext"));
+				break;
+			case FAIL_CANCELLED:
+				Operators.onDiscard(value, currentContext());
+				break;
+			case FAIL_TERMINATED:
+				Operators.onNextDropped(value, currentContext());
+				break;
+			case OK:
+				break;
+		}
+	}
+
+	@Override
+	public Emission tryEmitNext(T t) {
+		Objects.requireNonNull(t, "t");
+
+		switch (state) {
+			case INITIAL:
+				// TODO different Emission?
```

Comment:
I don't think we need a different emission for `tryEmit[Error|Complete]`, but for `tryEmitNext` we might, so we can distinguish early attempts at emitting through a `Sink` that doesn't "warm up" from `FAIL_OVERFLOW` (and `emitNext` would also discard in these cases)... follow up issue I think.